src/hyperv/hyperv_api_v1.c

```diff
@@ -1663,13 +1671,11 @@ hyperv1DomainAttachSyntheticEthernetAdapter(virDomainPtr domain,
     virtualSystemIdentifiers = virBufferContentAndReset(&query);
 
     /* build the __PATH variable of the switch port */
-    if (hyperv1MsvmComputerSystemFromDomain(domain, &computer) < 0)
-        goto cleanup;
     if (virAsprintf(&connection__PATH, "\\\\%s\\root\\virtualization:"
                 "Msvm_SwitchPort.CreationClassName=\"Msvm_SwitchPort\","
                 "Name=\"%s\",SystemCreationClassName=\"Msvm_VirtualSwitch\","
-                "SystemName=\"%s\"", computer->data->ElementName,
-                switchport_guid_string, net->data.network.name) < 0)
+                "SystemName=\"%s\"", hostname,
```

Comment:
the hostname is passed from hypervNDomainDefineXML and is result of hypervNGetHostSystem call which in turn is Msvm_ComputerSystem->data->ElementName so we had 2 lookups here instead of using what we already have given to us by the caller.